ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/WrappingRule.kt

```diff
@@ -529,6 +533,23 @@ public class WrappingRule :
         }
     }
 
+    private fun insertNewLineBeforeSemi(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
+    ) {
+        val previousCodeLeaf = node.prevCodeLeaf()?.lastChildLeafOrSelf() ?: return
+        val nextCodeLeaf = node.nextCodeLeaf()?.firstChildLeafOrSelf() ?: return
+        if (previousCodeLeaf.treeParent.elementType == ENUM_ENTRY && nextCodeLeaf.elementType == RBRACE) {
+            // Allow
+            // enum class INDEX2 { ONE, TWO, THREE; }
+            return
+        }
+        if (noNewLineInClosedRange(previousCodeLeaf, nextCodeLeaf)) {
+            requireNewlineAfterLeaf(node, autoCorrect, emit, indent = previousCodeLeaf.indent())
```

Comment:
The method is called `insertNewLineBeforeSemi` but actually the newline is added after the semicolon. The latter is preferred, because it will produce the best result when the `no-semi` rule is disabled.

Reply:
By `latter` if you mean new line after semi then yes that is also my understanding. I will update the name to `insertNewLineAfterSemi`